### PR TITLE
fix(abacus): fix bug in read LATTICE_CONSTANT in md

### PR DESCRIPTION
**Pull request by pxlxingliang**
In ABACUS version >= v3.1.4, the format of file "MD_dump" is different. The lattice constant line is added by the unit "Angstrom" at the end, and the position is the real cartesian coordinate. 
MD_dump is changed from 
```
MDSTEP:  0
LATTICE_CONSTANT: 23.453780000000
LATTICE_VECTORS
  1.000000000000  0.000000000000  0.000000000000
  0.000000000000  1.000000000000  0.000000000000
  0.000000000000  0.000000000000  1.000000000000
VIRIAL (KBAR)
  36.689617311102  0.000000000001  -0.000000000000
  0.000000000001  36.689617311104  -0.000000000001
  -0.000000000000  -0.000000000001  36.689617311147
INDEX    LABEL    POSITIONS    FORCE (eV/Angstrom)
  0  Sn  0.000000000000  0.000000000000  0.000000000000  -0.000000000000  0.000000000000  -0.000000000000
  1  Sn  0.000000000000  0.250000000000  0.250000000000  -0.000186795145  -0.000453823732  -0.000453823732
```

to

```
MDSTEP:  0
LATTICE_CONSTANT: 12.411200939060 Angstrom
LATTICE_VECTORS
  1.000000000000  0.000000000000  0.000000000000
  0.000000000000  1.000000000000  0.000000000000
  0.000000000000  0.000000000000  1.000000000000
VIRIAL (kbar)
  36.689617311101  0.000000000003  -0.000000000000
  0.000000000003  36.689617311089  -0.000000000001
  -0.000000000000  -0.000000000001  36.689617311135
INDEX    LABEL    POSITION (Angstrom)    FORCE (eV/Angstrom)    VELOCITY (Angstrom/fs)
  0  Sn  0.000000000000  0.000000000000  0.000000000000  -0.000000000000  -0.000000000001  -0.000000000001  0.001244557166  -0.000346684288  0.000768457739
  1  Sn  0.000000000000  3.102800034079  3.102800034079  -0.000186795145  -0.000453823768  -0.000453823768  0.000550996187  -0.000886442775  0.001579501983
```

### CODE REVIEW

**Review by njzjz — APPROVED** [review]

**Comment by pxlxingliang** [comment]
@njzjz fix the bug in reading  lattice constant and coordinate, and add unittest for new version.

File: dpdata/abacus/md.py [hunk 1]
```diff
@@ -42,6 +41,36 @@ def get_coord_dump_freq(inlines):
...
+      0.000000000000  1.000000000000  0.000000000000
+      0.000000000000  0.000000000000  1.000000000000
+    VIRIAL (KBAR)
+      36.689617311101  0.000000000003  -0.000000000000
+      0.000000000003  36.689617311089  -0.000000000001
+      -0.000000000000  -0.000000000001  36.689617311135
+    INDEX    LABEL    POSITIONS    FORCE (eV/Angstrom)
+      0  Sn  0.000000000000  0.000000000000  0.000000000000  -0.000000000000  -0.000000000001  -0.000000000001
+      1  Sn  0.000000000000  0.250000000000  0.250000000000  -0.000186795145  -0.000453823768  -0.000453823768
+    """
```
Comment (njzjz) [inline dpdata/abacus/md.py:73]: We follow [numpydoc](https://numpydoc.readthedocs.io/en/latest/format.html) to use docstring as documentation. Please do not leave raw data in the docstring.

**Review by njzjz — CHANGES_REQUESTED** [review]

File: dpdata/abacus/md.py (hunk 1 above)
Comment (pxlxingliang) [reply dpdata/abacus/md.py]: Ok, it is removed.

**Review by njzjz — APPROVED** [review]

**Comment by AnguseZhang** [comment]
@pxlxingliang Is this patch compatible with the version older than 3.1.4?

**Comment by pxlxingliang** [comment]
Yes，it is compatible发自我的手机-------- 原始邮件 --------发件人： Yuzhi Zhang ***@***.***>日期： 2023年4月22日周六 10:52收件人： deepmodeling/dpdata ***@***.***>抄送： pxlxingliang ***@***.***>, Mention ***@***.***>主    题： Re: [deepmodeling/dpdata] fix(abacus): fix bug in read LATTICE_CONSTANT in md (PR #450)
@pxlxingliang Is this patch compatible with the version older than 3.1.4?

**Comment by pxlxingliang** [comment]
@njzjz @wanghan-iapcm，please help check this PR. If there is no problem, please help merge it as soon as possible. This bug has a significant effect on the using of dpdata+ABACUS. Besides, I hope a new version to be relased after this PR is merged.

**Pull request merged**